### DISCUSSION TITLE
Can I use my own dataset for this???

### DISCUSSION THREAD

**Discussion by aniket2666** [AlexeyAB/darknet#8624] [Q&A]
Hey Alex,
I wanna know if I can train the model using my own dataset.
If yes, please can you spare some time and guide me a bit where and how I can upload my own dataset and train the model for the same?

Really eager to know bout it, waiting for your response

**Comment by DieBirke** [comment]
Hello, yes you can make/use your own dataset (I recommend using labelImg https://github.com/heartexlabs/labelImg to draw bounding boxes which coordinates are then automatically converted to the yolo format)

You have to go through the installation process to run the algorithm on your own device.

Another popular option is to use google colab. There are tutorials with ready to use notebooks where you just upload your dataset to your google drive which then is accesed through google colab to train and detect.

I can not give a throughout guide here but there are a lot of tutorials out there.

**Comment by aniket2666** [comment]
Thanks, I already have a dataset and I have the images labelled in yolo
format. I just don't know how to fetch them and use them in your algorithm.
Nevertheless, I'll try my best to make full use of your algorithm. Thanks a
lot

**Reply by DieBirke** [reply]
You specify the location of your dataset entries in a .txt file. Then you specify the location of the .txt file in the obj.data file.
Something like this:

**obj.data file**
classes = 2
train = [PATH]/train.txt
valid =  [PATH]/test.txt
names = [PATH]/obj.names
backup =  [PATH]/backup/

**train.txt file**
[PATH]/image0.jpg
[PATH]/image1.jpg
...
